include/deal.II/base/vectorization.h

```diff
@@ -1096,6 +1096,15 @@ class VectorizedArray<double, 8>
     data = _mm512_loadu_pd(ptr);
   }
 
+  DEAL_II_ALWAYS_INLINE
+  void
+  load(const float *ptr)
+  {
+    DEAL_II_OPENMP_SIMD_PRAGMA
+    for (unsigned int i = 0; i < 8; ++i)
+      data[i] = ptr[i];
+  }
```

Comment:
@kronbichler Thank you for the code. Indeed, the code is now somewhat faster, since nearly 190GB/sec is reached. I have seen similar structures elsewhere but decided against for now due to the next point and the variable number of blocks (which is now fixed by setting its value to 128).

A remaining question is how to deal with mixed types (temporal vector is double and global vectors are float): as far as I know we do not have a way to cast from `VectorizedArray<float, VectorizedArray<double>::size()>` to `VectorizedArray<double>`.  To get the code running, I have create a new `load`/`store` function. I didn't find appropriate intrinsic instructions (cast/broadcast); hopefully `DEAL_II_OPENMP_SIMD_PRAGMA` does the job. Do you agree with this approach?

Reply:
I agree with the `load` function. There is a way to address this with intrinsics (I did not test whether the compiler-generated code is bad), which is for the example of `float` -> `double` conversion for AVX512:
```
    __m256 tmp = _mm256_loadu_ps(ptr);
    data = _mm512_cvtps_pd(tmp);
```
There are similar operations for the other way around and for the different AVX/SSE types. This does not need to happen now, as long as we remember to look into the code generated by these functions and compare with optimal code.